test/distributed/fsdp/test_fsdp_grad_acc.py

```diff
@@ -197,7 +207,11 @@ def permute_tensor(x: torch.Tensor):
             loss.backward()
             losses.append(loss)
             acc_loss = sum(losses)
-            acc_grads = [p.grad.detach().clone() for p in fsdp_model.parameters()]
+            acc_grads = [
+                p.grad.detach().clone()
+                for p in fsdp_model.parameters()
+                if p.grad is not None
```

Comment:
`use_orig_params=True` will always require this idiom (for better or for worse) because even though an original parameter requires gradient, it may not receive a gradient if it is not in the local shard. I think this design matches the SPMD spirit best, but it just requires always having this `if grad is not None`.